### DISCUSSION TITLE
Want to run CKAN on apache2 on 5000 port, but while starting apache2, it is returning error "could not bind to address 0.0.0.0:5000".

### DISCUSSION THREAD

**Discussion by ArqamFarooqui110719** [ckan/ckan#6468] [Q&A] [answered]
CKAN      : v2.9.3(docker)
Apache2 : v2.4.41

We have pulled CKAN-2.9.3 image. After installing the apache2, supervisor services etc in the CKAN container. we are trying to run CKAN on Apache2 by giving the port `LISTEN 5000` in `/etc/apache2/port.conf` file and using below configurations in `/etc/apache2/site-enabled/ckan_default.conf` file for `ckan_default` process:
```
<VirtualHost *:5000>
     ServerName localhost
     ServerName localhost
     WSGIScriptAlias /ckan /etc/ckan/ckan.wsgi
     WSGIScriptAlias /     /etc/ckan/ckan.wsgi

    # Pass authorization info on (needed for rest api).
    WSGIPassAuthorization On

    # Deploy as a daemon (avoids conflicts between CKAN instances).
    WSGIDaemonProcess ckan_default display-name=ckan_default user=www-data group=www-data home=/usr/lib/ckan/venv/src/ckan/ threads=15
    WSGIProcessGroup ckan_default

    ErrorLog  "|/usr/bin/rotatelogs -n 5 /var/log/apache2/ckan_http.error.log 100M"
    CustomLog "|/usr/bin/rotatelogs -n 5 /var/log/apache2/ckan_http.custom.log 100M" combined
    
    <Directory />
        Require all granted
    </Directory>

</VirtualHost>
```
After these changes when we are starting the Apache2 service, we are getting PORT conflict error i.e. 

```
(98)Address already in use: AH00072: make_sock: could not bind to address [::]:5000
(98)Address already in use: AH00072: make_sock: could not bind to address 0.0.0.0:5000
no listening sockets available, shutting down
AH00015: Unable to open logs
Action 'start' failed.
The Apache error log may have more information.
```

**Comment by frafra** [comment]
You probably have CKAN already running and binding to port 5000. Look at your processes.

**Reply by ArqamFarooqui110719** [reply]
Thanks for your response @frafra, I have resolved the issue.

**Comment by ArqamFarooqui110719** [comment, marked as answer]
I have resolved the above issue itself by installing the apache2,supervisor etc. services via Dockerfile itself.
For that I have updated the Dockerfile with the required packages and created the Image. and I have also put the required files( like, apache2's port.conf, ckan_default,conf etc.) of the respective packages on a mounted volume path.

For apache2:
/etc/apache2/ -> /path/apache2/
(above apache2 path should contain all required and updated apache2 files)
